### DISCUSSION TITLE
How can I handle a multi-timeseries dataset?

### DISCUSSION THREAD

**Discussion by anhvung** [timeseriesAI/tsai#868] [General]
Can tsai's models handle datasets with multiple timeseries? For example I want to forecast the demand of different products individually. If there are 100 products, there will be 100 timeseries to forecast. 
Some brands will have similar behavior and others will differ depending on the features, that's why I want to train one model instead of 100 models for each product.
The number of rows of the dataframe would be nbr_product * nbr_features * timeseries_length

**Comment by MOREDataset** [comment]
Hi @anhvung,

I don't think your solution is possible. However, you can either, as you said, develop a 100 forecasting models one for each product, or I think you can approach this problem as a multi-ouput regression problem. I am trying to figure out how to do something similar myself, but it seems that to have multiple outputs the head needs to be defined manually. Not sure how though! Maybe @oguiza can help with a simple example?

Let me know if you found anything helpful as it has been some time since you posted this...
Much luck!
